### DISCUSSION TITLE
Support Immax NEO Smart Sirene (TuYa Device)

### DISCUSSION THREAD

**Discussion by larsmeyer112** [Koenkk/zigbee2mqtt#21878] [Devices]
Hi,
I am trying to integrate the Immax NEO Smart Sirene Zigbee 3.0 (07504L / MS-S7B-G) without success because I have problems understanding how to create specific exposes.

As an example here is one of the datapoints Immax told me:

- DP ID / DP name / identifier / data transfer type / data type / properties
- 4 / alert / alert_state / Send and Report / Bool / Edit, Delete

Here ist my code:

// ms-s7b-g.js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
    // Tuya Fingerprint
    fingerprint: [
        {
            modelID: 'TS0219',
            manufacturerName: '_TYZB01_b6eaxdlh',
        },
    ],
    model: 'MS-S7B-G',
    vendor: 'Immax NEO',
    description: 'Smart Sirene Zigbee 30 - 07504L',
    fromZigbee: [tuya.fz.datapoints],
    toZigbee: [tuya.tz.datapoints],
    onEvent: tuya.onEventSetTime,
    configure: tuya.configureMagicPacket,
    // functionality
	exposes: [
          e.binary('alert_state', ea.STATE_SET, 'ON', 'OFF').withDescription('Turn the alarm ON/OFF'),
    ],
    meta: {
        // datapoints
		tuyaDatapoints: [
            [4, 'alert_state', tuya.valueConverter.onOff],
        ],
    },
};
module.exports = definition;

------------

I always get the message, that there is "no converter available for 'alert state'", do I have to use an existing definition instead of 'alert_state' or define it manually?

Is there a guide how to handle a datepoint that is not using a standard expose like humidity, that is usesed in the section "Support new TuYa devices" on the Zigbee2MQTT website?

Thanks for any help!

Kind Regards
Lars

**Comment by larsmeyer112** [comment]
I am still trying to integrate the device, maybe someone can help me ...

Is there an "easy" way to integrate it by using the following converter?

No converter available for 'MS-S7B-G' with cluster 'ssIasZone' and type 'commandStatusChangeNotification' and data '{"extendedstatus":0,"zonestatus":16}'

**Comment by cdalexndr** [comment]
See also https://github.com/Koenkk/zigbee-herdsman-converters/pull/9005
